### PR TITLE
VAE

### PR DESCRIPTION
**Pull request by rijobro**
Ports VAE functionality from ptproto.

### Status
**Work in progress**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [x] Integration tests passed locally by running `./runtests.sh --codeformat --coverage`.
- [x] In-line docstrings updated.
- [x] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by rijobro** [comment]
Complementary notebook: https://github.com/Project-MONAI/tutorials/pull/68.

File: monai/networks/nets/varautoencoder.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        x = x.view(x.shape[0], -1)
+        mu = self.mu(x)
+        logvar = self.logvar(x)
+        return mu, logvar
+
+    def decode_forward(self, z: torch.Tensor) -> torch.Tensor:
+        x = F.relu(self.decodeL(z))
+        x = x.view(x.shape[0], self.channels[-1], *self.final_size)
+        x = self.decode(x)
+        x = torch.sigmoid(x)
```
Comment (wyli) [inline monai/networks/nets/varautoencoder.py:88]: could you make this sigmoid optional?

File: monai/networks/nets/varautoencoder.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+import torch.nn as nn
+from torch.nn import functional as F
+
+from monai.networks.layers.convutils import calculate_out_shape, same_padding
+from monai.networks.layers.factories import Act, Norm
+from monai.networks.nets import AutoEncoder
+
+
+class VarAutoEncoder(AutoEncoder):
+    def __init__(
```
Comment (wyli) [inline monai/networks/nets/varautoencoder.py:25]: please put the paper link https://arxiv.org/abs/1312.6114

**Review by wyli — APPROVED** [review]
thanks I have some minor comments inline

**Comment by rijobro** [comment]
@wyli updated as per your comments. Thanks.
